managers/s3.py

```diff
@@ -37,11 +42,49 @@ def createS3Bucket(self, bucketName, bucketPermissions):
         raise S3Error('Unable to create bucket in s3')
 
 
-    def createManifestInS3(self, manifestPath, manifestJSON, s3_bucket: str):
+    def store_pdf_manifest(self, record, bucket_name, flags=FileFlags(reader=True), path: str=None):
+        record_id = record.source_id.split('|')[0]
+        pdf_part = next(filter(lambda part: part.file_type == 'application/pdf', record.get_parts()), None)
+
+        if pdf_part is not None:
+            if path:
+                manifest_path = f'{path}/{record.source}/{record_id}.json'
+
+            print(manifest_path)
+            manifest_url = get_stored_file_url(storage_name=bucket_name, file_path=manifest_path)
+            manifest_json = self.generate_manifest(record=record, source_url=pdf_part.url, manifest_url=manifest_url)
+
+            self.create_manifest_in_s3(manifest_path=manifest_path, manifest_json=manifest_json, s3_bucket=bucket_name)
+
+            print(json.dumps(dataclasses.asdict(flags)))
+            record.has_part.insert(0, Part(
+                index=pdf_part.index,
+                url=manifest_url,
+                source=record.source,
+                file_type='application/webpub+json',
+                flags=json.dumps(dataclasses.asdict(flags))
+            ).to_string())
```

Comment:
Nice - it's looking great!

Let's just remove the print statements. 

You could simplify the path prefix param and just pass in if there's an additional path prefix after the manifests folder. This would ensure all manifests are always stored in the manifests folder in S3.

Reply:
Ok that makes sense with the path prefix